contractcourt/channel_arbitrator.go

```diff
@@ -3159,3 +3252,71 @@ func (c *ChannelArbitrator) checkLegacyBreach() (ArbitratorState, error) {
 	// This is a modern breach close with resolvers.
 	return StateContractClosed, nil
 }
+
+// cancelIncomingHTLCs cancels back the incoming htlcs for the corresponding
+// outgoing htlc. An example could be dust htlcs which cannot be enforced
+// onchain therefore we can immediately cancel back the corresponding incoming
+// htlc even if the commitment tx is still not confirmed and there is a risk of
+// another state confirming e.g. the remote pending commitment. However we
+// might in the future cancel back non-dust HTLCs that's why this function is
+// kept generic and not only for dust HTLCs.
+func (c *ChannelArbitrator) cancelIncomingHTLCs(
+	htlcs fn.Set[uint64]) error {
+
+	var msgsToSend []ResolutionMsg
+	failureMsg := &lnwire.FailPermanentChannelFailure{}
+
+	for idx := range htlcs {
+		failMsg := ResolutionMsg{
+			SourceChan: c.cfg.ShortChanID,
+			HtlcIndex:  idx,
+			Failure:    failureMsg,
+		}
+
+		msgsToSend = append(msgsToSend, failMsg)
+	}
+
+	// Send the msg to the switch.
+	if len(msgsToSend) == 0 {
+		return nil
+	}
+
+	log.Debugf("ChannelArbitrator(%v): sending "+
+		"resolution message=%v", c.cfg.ChanPoint,
+		lnutils.SpewLogClosure(msgsToSend))
+
+	err := c.cfg.DeliverResolutionMsg(msgsToSend...)
+	if err != nil {
+		log.Errorf("unable to send pkts: %v", err)
```

Comment:
```suggestion
		log.Errorf("Unable to send pkts: %v", err)
```

Reply:
maybe we could define an error `ErrHTLCIDNotFound` and return it here,
https://github.com/lightningnetwork/lnd/blob/0dd58ee52937a545fd82108678e9a31611b7471f/htlcswitch/switch.go#L1341-L1347

since it's expected to have this error here if we cancel twice, we could instead log a warning if `errors.Is(err, ErrHTLCIDNotFound)` given it's a dust htlc.

Reply:
going to remove the db entry and add the error.